dao/FashionKnex.js

```diff
@@ -0,0 +1,75 @@
+const knex = require('knex')
+const Model = require('fashion-model/Model')
+const conflogger = require('conflogger')
+const _getTableName = require('./util/getTableName')
+
+function _clean (document) {
+  return Model.isModel(document)
+    ? Model.clean(document)
+    : document
+}
+
+module.exports = class FashionKnex {
+  constructor (options) {
+    let {
+      modelType,
+      tableName,
+      logger,
+      knexConfig
+    } = options
+
+    this._logger = conflogger.configure(logger)
+    this._modelType = modelType
+    this._tableName = tableName || _getTableName(modelType)
+    this._knex = knex(knexConfig)
+  }
+
+  async insert (data, returning) {
+    data = _clean(data)
```

Comment:
This is a bit tricky since fashion-model is fairly different from knex. I think that as long as the the initial table is created with the correct data types and consumers of this API are passing fashion-models instead of raw data, it should be fine because fashion-model should be doing the validation. I do have some ideas for validation, particularly in creating the table using an initial provided model, but I think I should just get to this in a separate PR.